### ISSUE TITLE
Creating windows pods flaky due to HNS error "failed to create the new HNSEndpoint: hnsCall failed in Win32: The object already exists. (0x1392)"

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#74766]
**What happened**:
Environment: On GCE with the win-bridge cni plugin

On a node when creating a pod for the first time, running podsandbox would fail with the error message below. "failed to create the new HNSEndpoint: hnsCall failed in Win32: The object already exists. (0x1392)"

```
Events:
  Type     Reason                  Age                From                                              Message
  ----     ------                  ----               ----                                              -------
  Warning  FailedCreatePodSandBox  15s                kubelet, e2e-test-yjhong-windows-node-group-rdt6  Failed create pod sandbox: rpc error: code = Unknown desc = [failed to set up sandbox container "5a4a06b6983a52f678e15203f1e77e0e310933e5a506dc3167bf7f3e57782106" network for pod "shetland": NetworkPlugin cni failed to set up pod "shetland_default" network: error while ProvisionEndpoint(5a4a06b6983a52f678e15203f1e77e0e310933e5a506dc3167bf7f3e57782106_l2bridge,359CAE70-8CA4-4143-A42B-42644CF04022,5a4a06b6983a52f678e15203f1e77e0e310933e5a506dc3167bf7f3e57782106): failed to create the new HNSEndpoint: hnsCall failed in Win32: The object already exists. (0x1392), failed to clean up sandbox container "5a4a06b6983a52f678e15203f1e77e0e310933e5a506dc3167bf7f3e57782106" network for pod "shetland": NetworkPlugin cni failed to teardown pod "shetland_default" network: failed to find HNSEndpoint 5a4a06b6983a52f678e15203f1e77e0e310933e5a506dc3167bf7f3e57782106_l2bridge: Endpoint 5a4a06b6983a52f678e15203f1e77e0e310933e5a506dc3167bf7f3e57782106_l2bridge not found]
  Normal   SandboxChanged          12s (x2 over 14s)  kubelet, e2e-test-yjhong-windows-node-group-rdt6  Pod sandbox changed, it will be killed and re-created.

```
Subsequent retry would succeed, and launching other pods afterwards would also, generally, be fine. 

I've found https://github.com/Microsoft/SDN/issues/298, but in that issue, the creation failed permanently.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): master (1.14, da5fd93e93f4)
- Cloud provider or hardware configuration: GCE
- OS (e.g: `cat /etc/os-release`): Windows 1809
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/cc @pjh @kubernetes/sig-windows-bugs

**Comment by michmike** [comment]
cc: @PatrickLang

**Comment by daschott** [comment]
@dineshgovindasamy @nagiesek We've seen this before -- what's the root-cause?

**Comment by dineshgovindasamy** [comment]
So this happens only for the first time right?

@yujuhong Could you please collect the following logs for us?

On the node before creating the pod for the first time.

start-birstranfer https://raw.githubusercontent.com/Microsoft/SDN/master/Kubernetes/windows/debug/collectlogs.ps1

run collectlogs.ps1

then start the trace by running the following command

C:\k\debug\starthnstrace.cmd

repro the issue

run "netsh trace stop"

then do again collectlogs.ps1

and send us both before and after collectlogs.ps1 and C:\server.etl

cc @nagiesek @madhanrm @daschott

**Comment by pjh** [comment]
@dineshgovindasamy thank you for the detailed steps to capture the traces. I've uploaded the trace + log files at https://github.com/pjh/kubernetes/tree/debugging/traces/74766_0. Files from both `collectlogs.ps1` and `cluster/log-dump/log-dump.sh` are included.

There was about a ten-minute delay between the node joining the cluster and the deployment of a pod to start the repro. The timestamp on the node just before creating the pod was Friday, March 1, 2019 6:56:06 PM.

Please let us know if you have questions or need more information. Thanks for taking a look!

**Comment by dineshgovindasamy** [comment]
Thanks @yujuhong @pjh. Logs were super useful in root causing the issue.

You are using L2Bridge in host gateway mode, so u r allocating .2 ip as the Gateway Ip for this host.

I am seeing a request to create an endpoint with an IP "10.64.1.2"

"{"Name":"1885009e399aa542ad60bab063262196a804252010ab401967a6575cc7ccd1bd_l2bridge","VirtualNetwork":"64736A7F-E9F0-4465-B861-7879393DDD22","Policies":[{"ExceptionList":["10.64.0.0/14","10.0.0.0/16","10.40.0.0/22"],"Type":"OutBoundNAT"},{"DestinationPrefix":"10.0.0.0/16","NeedEncap":true,"Type":"ROUTE"},{"DestinationPrefix":"10.40.0.3/32","NeedEncap":true,"Type":"ROUTE"}],"IPAddress":"10.64.1.2","DNSSuffix":"cluster.local","DNSServerList":"10.0.0.10","GatewayAddress":"10.64.1.2"}"


But this IP is assigned to CBR0(Gateway) and an endpoint already exists on HNS with that IP.

so when a container request is coming with the same IP we fail saying "the object already exists in our Address MAP" 

Now the question is why are we creating a container with an IP 10.64.1.2 when that IP is allocated for the Gateway NIC on this host?

u can see that in Before logs on HNS -> endpoint.txt

ID                                   IPAddress    MacAddress        IsRemoteEnd

                                                                   point

--                                   ---------    ----------        -----------

**428D83D1-6138-459F-9B77-4F8B48A6FF6B 10.64.1.2**    00-15-5D-6E-A4-DD

First Entry is the endpoint that already exists

but u are also sending a request to create a container with the same IP

So basically ur IPAM is not excluding the gateway NIC and hence we are seeing that only for the first time u create the container. Hope this helps in root causing the issue on ur end.

Again thanks for the logs. Super Useful..
